src/aux-records/XpController.ts

```diff
@@ -0,0 +1,711 @@
+                errorCode: 'invalid_request',
+                errorMessage:
+                    'The issuing and receiving party cannot be the same user.',
+            };
+
+        return { success: true, receivingUser, issuingUser };
+    }
+
+    /**
+     * Get an Xp user's meta data (Xp meta associated with an auth user)
+     * Creates an Xp user for the auth user if one does not exist
+     */
+    async getXpUser(id: GetXpUserById): Promise<GetXpUserResult> {
+        return await tryScope(
+            async () => {
+                let user =
+                    (await ('userId' in id
+                        ? this._xpStore.getXpUserByAuthId(id.userId)
+                        : this._xpStore.getXpUserById(id.xpId))) ?? null;
+                if (!user && 'userId' in id) {
+                    const result = await this._createXpUser(id.userId);
+                    if (result.success) {
+                        user = result.user;
+                    } else return result;
+                }
+                return user
+                    ? { success: true, user }
+                    : {
+                          success: false,
+                          errorCode: 'user_not_found',
+                          errorMessage: 'The user was not found.',
+                      };
+            },
+            {
+                scope: [TRACE_NAME, this.getXpUser.name],
+                errMsg: 'An error occurred while getting the user.',
+                returnOnError: {
+                    success: false,
+                    errorCode: 'server_error',
+                    errorMessage: 'An error occurred while getting the user.',
+                },
+            }
+        );
+    }
+
+    async batchGetXpUsers(
+        queryOptions:
+            | {
+                  xpId: XpUser['id'][];
+                  authId?: AuthUser['id'][];
+              }
+            | {
+                  authId: AuthUser['id'][];
+                  xpId?: XpUser['id'][];
+              }
+    ): Promise<BatchGetXpUserResult> {
+        return await tryScope(
+            async () => {
+                const results = await this._xpStore.batchQueryXpUsers(
+                    queryOptions
+                );
+                return {
+                    success: true,
+                    users: results,
+                };
+            },
+            {
+                scope: [TRACE_NAME, this.batchGetXpUsers.name],
+                errMsg: 'An error occurred while getting the users.',
+                returnOnError: {
+                    success: false,
+                    errorCode: 'server_error',
+                    errorMessage: 'An error occurred while getting the users.',
+                },
+            }
+        );
+    }
+
+    /**
+     * Get a contract by its id
+     * @param id The id of the contract to get
+     */
+    async getContractById(id: XpContract['id']): Promise<GetContractResult> {
```

Comment:
@LandonSiler This function should also accept the current user ID so that we can validate whether the current user is allowed to access the given contract.